cmd/es-rollover/app/actions.go

```diff
@@ -49,16 +49,18 @@ type ActionCreatorFunction func(client.Client, Config) Action
 // ExecuteAction execute the action returned by the createAction function
 func ExecuteAction(opts ActionExecuteOptions, createAction ActionCreatorFunction) error {
 	cfg := Config{}
-	cfg.InitFromViper(opts.Viper)
-	tlsOpts, err := opts.TLSFlags.InitFromViper(opts.Viper)
-	if err != nil {
-		return err
+	if err := cfg.InitFromViper(opts.Viper); err != nil {
+		opts.Logger.Error("Failed to initialize config from viper",
+			zap.Error(err),
+		)
```

Comment:
there is no need (ever) to log errors if you are returning it anyway - somewhere upstream the caller will decide to log it.